source/common/common/envoy_defaults.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "source/common/singleton/threadsafe_singleton.h"
+
+namespace Envoy {
+
+struct DefaultsProfile {
+  struct Cluster {
+    uint32_t max_buffer_size = 32768;
+  };
+
+  struct Http2 {
+    uint32_t initial_connection_window_size = 1048576; // 1 MiB
+    uint32_t max_concurrent_streams = 100;
+    uint32_t initial_stream_window_size = 65536; // 64 KiB
+  };
+
+  struct HttpConnectionManager {
```

Comment:
I'm not sure if I agree with the direction this is going. I strongly suggest expressing defaults in proto form so that we can automatically generate documentation from them and also so that they can be represented in a manner consistent with the actual config structure.

This is similar to where I was heading with the manifest file, https://github.com/envoyproxy/envoy/blob/main/docs/protodoc_manifest.yaml. We don't have to do this exactly here, but I think we do want to go with something proto based.

CC @phlax @mattklein123

Reply:
I haven't reviewed this PR (LMK if you want me to) but big +1. The safe defaults must be defined in the protos, hopefully as annotations so they can be easily pulled into code/docs.

Reply:
Yes, we will be getting overrides from manifest. @acsaeed started working on it.